src/cli/peakdetector/PeakDetectorCLI.cpp

```diff
@@ -663,35 +663,32 @@ void PeakDetectorCLI::saveCSV(string setName) {
     #endif
 
 	string fileName = mavenParameters->outputdir + setName + ".csv";
-
-    CSVReports* csvreports = new CSVReports(mavenParameters->samples);
-    csvreports->setMavenParameters(mavenParameters);
-
+	if(fileName.empty()) return;
+	
```

Comment:
fileName would never be empty. Use this instead:
if (mavenParameters->outputdir.empty()) return;
string fileName = mavenParameters->outputdir + setName + ".csv";